### DISCUSSION TITLE
brew install python failing

### DISCUSSION THREAD

**Discussion by h-walter** [Homebrew/discussions#4927] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: >=2.5.0 (shallow or no git repository)
ORIGIN: https://github.com/h-walter/brew.git
HEAD: 251a098fbf1c02ca95f53b4806e5b6da8d5e4a80
Last commit: 4 hours ago
Core tap JSON: 26 Nov 02:22 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BREW_GIT_REMOTE: https://github.com/h-walter/brew.git
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/h-walter/homebrew-core.git
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1.1-arm64
CLT: 15.0.0.40.1
Xcode: 15.0.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

When trying to install python im getting this error 
`Error: undefined method `[]' for nil:NilClass`

**Comment by Bo98** [comment, marked as answer]
Looks like something's wrong with your git clone:

```
HOMEBREW_VERSION: >=2.5.0 (shallow or no git repository)
```

**Reply by h-walter** [reply]
Thanks! How can I fix that?

**Reply by Bo98** [reply]
I'm not sure. Does `brew update` output any errors?

**Reply by Bo98** [reply]
You can also try `git -C "$(brew --repo)" fetch --unshallow`

**Reply by h-walter** [reply]
Nope no errors at all
`/opt/homebrew/bin/brew update-report --debug
Already up-to-date.
return 0
exit 0`
same with upgrade

also: git -C "$(brew --repo)" fetch --unshallow
`fatal: --unshallow on a complete repository does not make sense`

**Reply by Bo98** [reply]
Ah right the issue is because you have set `brew` to use your own fork:

```
HOMEBREW_BREW_GIT_REMOTE: https://github.com/h-walter/brew.git
```

but that fork doesn't have any tags:

<img width="97" alt="Screenshot 2023-11-26 at 03 06 39" src="https://github.com/Homebrew/discussions/assets/1190754/4f37239f-64da-4102-9f82-a5092a4f42de">

Is there a reason you've set it to your own fork? If it's intentional: you'll need to regularly keep the tags in sync.

(Technically speaking, it shouldn't be erroring entirely if it can't detect a tag, but I'll fix that soon. It's still worth fixing `HOMEBREW_VERSION` though as some other things do use it.)

**Reply by h-walter** [reply]
I just used my own fork because i thought it said so on the install page brew 
Is there another way to install without fork?
https://docs.brew.sh/Installation

**Reply by Bo98** [reply]
You don't need any fork. That page is perhaps unclear - I'll fix it.

Ignore `HOMEBREW_BREW_GIT_REMOTE` and `HOMEBREW_CORE_GIT_REMOTE` entirely - don't set them, run `brew update` and it should all work.

**Reply by h-walter** [reply]
Yep works perfectly Thanks!
